source/guide_up1.rst

```diff
@@ -0,0 +1,217 @@
+.. author:: ezra <ezra@posteo.de>
+.. highlight:: console
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/up1.png
+      :align: center
+
+###
+Up1
+###
+
+Up1_ provides a very simple web interface to share files end-to-end encrypted via an autogenerated url. It works without any registration or login, the encryption takes place in the clients browser, so the server will know nothing about the files content (see an example installation on on https://share.riseup.net).
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * Node.js_ and its package manager npm_
+  * supervisord_
+  * domains_
+
+Prerequisites
+=============
+
+We're using Node.js_ in the stable version 8:
+
+::
+
+ [isabell@stardust ~]$ uberspace tools version show node
+ Using 'Node.js' version: '8'
+ [isabell@stardust ~]$
+ 
+We will need to update the Node Packet Manager npm:
+
+::
+
+  [isabell@stardust ~]$ npm install npm@latest -g
+  + npm@6.4.1
+  updated 1 package in 14.629s
+  [isabell@stardust ~]$ 
+
+
+.. include:: includes/my-print-defaults.rst
+
+Your URL needs to be setup:
+
+.. include:: includes/web-domain-list.rst
+
+Installation
+============
+
+First get the Up1 source code from Github_:
+
+::
+
+  [isabell@stardust ~]$ git clone https://github.com/Upload/Up1 up1 --depth 1
+  Cloning into 'up1'...
+  remote: Enumerating objects: 37, done.
+  remote: Counting objects: 100% (37/37), done.
+  remote: Compressing objects: 100% (33/33), done.
+  remote: Total 37 (delta 0), reused 16 (delta 0), pack-reused 0
+  Unpacking objects: 100% (37/37), done.
+  Note: checking out '059f199a6f94fc6c90cde1e4dab0b5b85b49533c'.
+  [isabell@stardust ~]$
+
+You have now downloaded the master development branch, because this might behave like a beta version of the software, you should check out the latest release. Please be sure to replace the pseudo version number ``66.6`` here with the latest version number from the release feed_:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1
+  [isabell@stardust up1]$ git checkout v66.6 --depth1
+  You are in 'detached HEAD' state. You can look around, [...]
+  [isabell@stardust up1]$ git checkout -b latest
+  Switched to a new branch 'latest'
+  [isabell@stardust up1]$
+
+Then run npm_ to install all the dependencies:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1/server
+  [isabell@stardust server]$ npm install
+  [...]
+  added 131 packages from 90 contributors and audited 219 packages in 6.04s
+  found 17 vulnerabilities (4 low, 8 moderate, 5 high)
+  run `npm audit fix` to fix them, or `npm audit` for details
+  [isabell@stardust server]$ 
+
+You probably got the error that there are vulnerabilities to fix, so follow the instruction and run ``npm audit fix`` to install the latest package versions:
+
+::
+
+  [isabell@stardust ~]$ cd ~/up1/server
+  [isabell@stardust server]$ npm audit fix
+  [...]
+  fixed 17 of 17 vulnerabilities in 219 scanned packages
```

Comment:
```
[isabell@stardust server]$ npm audit fix

Usage: npm <command>

where <command> is one of:
    access, adduser, bin, bugs, c, cache, completion, config,
    ddp, dedupe, deprecate, dist-tag, docs, doctor, edit,
    explore, get, help, help-search, i, init, install,
    install-test, it, link, list, ln, login, logout, ls,
    outdated, owner, pack, ping, prefix, profile, prune,
    publish, rb, rebuild, repo, restart, root, run, run-script,
    s, se, search, set, shrinkwrap, star, stars, start, stop, t,
    team, test, token, tst, un, uninstall, unpublish, unstar,
    up, update, v, version, view, whoami

npm <command> -h     quick help on <command>
npm -l           display full usage info
npm help <term>  search for help on <term>
npm help npm     involved overview

Specify configs in the ini-formatted file:
    /home/isabell/.npmrc
or on the command line via: npm <command> --key value
Config info can be viewed via: npm help config

npm@5.6.0 /opt/nodejs8/lib/node_modules/npm

Did you mean this?
    edit
[isabell@stardust server]$
```

Reply:
Oh, first I thought its a mistake because you did not follow the guide and updatet to `npm 6.4.0`, but it looks like that the update does not refresh the current npm version. You have to start a new ssh session to use the updated npm version. 

Do you have any idea how to make the npm update effective without restarting the terminal?

Reply:
`hash -r` will force an update.

Reply:
ok nice